src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+                        case BinaryOperatorKind.LessThanOrEqual:
+                            shouldNegate = false;
+                            break;
+                        case BinaryOperatorKind.NotEquals:
+                        case BinaryOperatorKind.GreaterThan:
+                            shouldNegate = true;
+                            break;
+                        default:
+                            return false;
+                    }
+                    break;
+                case 1:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.LessThan:
+                            shouldNegate = false;
+                            break;
+                        case BinaryOperatorKind.GreaterThanOrEqual:
+                            shouldNegate = true;
+                            break;
+                        default:
+                            return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static bool IsRightCountComparison(IBinaryOperation binaryOperation, out bool shouldNegate)
+        {
+            shouldNegate = false;
+
+            if (!TryGetZeroOrOneConstant(binaryOperation.LeftOperand, out int constant))
+            {
+                return false;
+            }
+
+            switch (constant)
+            {
+                case 0:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.Equals:
+                        case BinaryOperatorKind.GreaterThanOrEqual:
+                            shouldNegate = false;
+                            break;
+
+                        case BinaryOperatorKind.LessThan:
+                        case BinaryOperatorKind.NotEquals:
+                            shouldNegate = true;
+                            break;
+
+                        default:
+                            return false;
+                    }
+                    break;
+                case 1:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.LessThanOrEqual:
+                            shouldNegate = true;
+                            break;
+
+                        case BinaryOperatorKind.GreaterThan:
+                            shouldNegate = false;
+                            break;
+
+                        default:
+                            return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static bool TypeContainsVsibileProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType propertyType)
```

Comment:
This seems like a very good candidate to become an extension method on `ITypeSymbol`:

```csharp
public static bool TypeContainsVisibleProperty(this ITypeSymbol type, string propertyName, SpecialType propertyType, INamedTypeSymbol contextSymbol)
```